### PR TITLE
Allow to upgrade to the global installed jhipster.

### PR DESCRIPTION
**Pull request by mshima**
Upgrade command only works from a released version to another released version making testing really difficult.
This PR allows to upgrade an app from a released version to the currently globally installed version.

-   Please make sure the below checklist is followed for Pull Requests.

-   [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [ ] Tests are added where necessary
-   [ ] Documentation is added/updated where necessary
-   [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

File: generators/upgrade/index.js [hunk 1]
```diff
@@ -468,14 +478,17 @@ module.exports = class extends BaseGenerator {
...
+            generateWithTargetVersion() {
                 const done = this.async();
                 this._cleanUp();
 
                 const blueprintInfo = this.blueprints
                     ? ` and ${this.blueprints.map(bp => bp.name + bp.latestBlueprintVersion).join(', ')} `
                     : '';
-                this._regenerate(this.latestJhipsterVersion, blueprintInfo, done);
+                const targetJhipsterVersion = this.originalTargetJhipsterVersion
+                    ? `${this.originalTargetJhipsterVersion} this.targetJhipsterVersion`
```
Comment (avdev4j) [inline generators/upgrade/index.js]: this.targetJhipsterVersion should be ${this.targetJhipsterVersion} ?

File: generators/upgrade/index.js [hunk 2]
```diff
@@ -468,14 +478,17 @@ module.exports = class extends BaseGenerator {
                 });
             },
 
-            generateWithLatestVersion() {
+            generateWithTargetVersion() {
                 const done = this.async();
                 this._cleanUp();
 
                 const blueprintInfo = this.blueprints
                     ? ` and ${this.blueprints.map(bp => bp.name + bp.latestBlueprintVersion).join(', ')} `
```
Comment (avdev4j) [inline generators/upgrade/index.js]: If we don't have a blueprint the message is weird 
> Successfully regenerated application with JHipster global this.targetJhipsterVersion and

File: generators/upgrade/index.js [hunk 3]
```diff
@@ -441,8 +448,11 @@ module.exports = class extends BaseGenerator {
             },
 
             updateJhipster() {
+                if (this.originalTargetJhipsterVersion === 'global') {
```
Comment (avdev4j) [inline generators/upgrade/index.js]: 'global' should be store into a class variable and documented as we can't know where this key is come from.

**Comment by avdev4j** [comment]
How I test it. 
Link a 6.6.0 JHipster version, generate a project with and change the JHipster (in the folder linked version) to a new one -> 6.7.1

Then I ran the following command:
`jhipster upgrade --target-version global`
```
INFO! Using JHipster version installed globally
INFO! Executing jhipster:upgrade
INFO! Options: targetVersion: global, from-cli: true, target-version: global
Welcome to the JHipster Upgrade Sub-Generator
This will upgrade your current application codebase to the latest JHipster version
No blueprints detected, skipping check of last blueprint version
Upgrading to the target JHipster version: 6.7.1
     info git rev-parse -q --is-inside-work-tree
```

Then the re-generation goes well but the npm install failed 
```
npm ERR! code ETARGET
npm ERR! notarget No matching version found for generator-jhipster@6.7.1
npm ERR! notarget In most cases you or one of your dependencies are requesting
npm ERR! notarget a package version that doesn't exist.
npm ERR! notarget
npm ERR! notarget It was specified as a dependency of 'test-module'
npm ERR! notarget

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/aviard/.npm/_logs/2020-01-14T15_59_25_933Z-debug.log
```

I think it's because the npm install command is looking for the 6.7.1 version that doesn't exist in the repository.  If we don't have a clean solution to avoid that, I think it will ok because this is an advanced feature for development purpose.

It works as expected and after fixing things from my review and discuss about the npm error log, we will able to merge it after resolved conflicts.

**Comment by avdev4j** [comment]
Oh and last but not least,
it deserve a documentation on the JHipster website to explain how to upgrade from a global version for developers ;)

**Comment by mraible** [comment]
Can you please rebase with master to fix conflicts?

**Comment by avdev4j** [comment]
For me changes are ok, we just need to fix eslint error and I'll merge this PR

**Comment by avdev4j** [comment]
good job @mshima 
thanks you

**Comment by mshima** [comment]
thank you @avdev4j for your help

**Pull request merged**